sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -2052,6 +2068,7 @@ object AnsiCast {
    * Spark's ANSI mode follows the syntax rules, except it specially allow the following
    * straightforward type conversions which are disallowed as per the SQL standard:
    *   - Numeric <=> Boolean
+   *   - Numeric <=> Timestamp
```

Comment:
shall we remove this change? By default we don't allow it in the ansi mode.

Reply:
done